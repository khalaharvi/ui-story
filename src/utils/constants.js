export const PRIMARY_CATEGORY_KEY = 'categories.lvl0';
export const SECONDARY_CATEGORY_KEY = 'categories.lvl1';
export const TERTIARY_CATEGORY_KEY = 'categories.lvl2';
