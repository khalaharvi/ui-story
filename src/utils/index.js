export { default as createClassNames } from './createClassNames';
export { default as getDisplayName } from './getDisplayName';
export { default as capitalizeFirstLetter } from './capitalizeFirstLetter';
export { default as slugify } from './slugify';
