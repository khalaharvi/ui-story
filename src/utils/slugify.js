export default (text) => {
  // Use hash map for special characters
  const specialChars = {"à":'a',"ä":'a',"á":'a',"â":'a',"æ":'a',"å":'a',"ë":'e',"è":'e',"é":'e', "ê":'e',"î":'i',"ï":'i',"ì":'i',"í":'i',"ò":'o',"ó":'o',"ö":'o',"ô":'o',"ø":'o',"ù":'o',"ú":'u',"ü":'u',"û":'u',"ñ":'n',"ç":'c',"ß":'s',"ÿ":'y',"œ":'o',"ŕ":'r',"ś":'s',"ń":'n',"ṕ":'p',"ẃ":'w',"ǵ":'g',"ǹ":'n',"ḿ":'m',"ǘ":'u',"ẍ":'x',"ź":'z',"ḧ":'h',"·":'-',"/":'-',"_":'-',",":'-',":":'-',";":'-'}; // eslint-disable-line

  return text.toString().toLowerCase()
    .replace(/\s+/g, '-') // Replace spaces with -
    .replace(/./g, target => specialChars[target] || target) // Replace special characters using the hash map
    .replace(/&/g, '-and-') // Replace & with 'and'
    .replace(/[^\w-]+/g, '') // Remove all non-word chars
    .replace(/--+/g, '-') // Replace multiple - with single -
    .replace(/^-+/, '') // Trim - from start of text
    .replace(/-+$/, ''); // Trim - from end of text
};
