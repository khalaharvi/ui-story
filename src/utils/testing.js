import { object } from 'prop-types';
import { channel, createBroadcast } from 'emotion-theming';
import { shallow, mount } from 'enzyme';
import theme from '@dering-hall/dh-theme';

const getOptions = (options = {}) => {
  const broadcast = createBroadcast(theme);

  return {
    context: {
      [channel]: broadcast
    },
    childContextTypes: {
      [channel]: object
    },
    ...options
  };
};

export const mountWithTheme = (
  Component,
  options = {}
) => mount(Component, { ...getOptions(options) });

export const shallowWithTheme = (
  Component,
  options = {}
) => shallow(Component, { ...getOptions(options) });

export default {
  mountWithTheme,
  shallowWithTheme
};
