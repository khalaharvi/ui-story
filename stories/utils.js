/* eslint-disable import/prefer-default-export */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { InstantSearch } from 'react-instantsearch/dom';
import { connectStateResults } from 'react-instantsearch/connectors';
import { withMedia, MediaQueryProvider } from 'react-media-query-hoc';
import { withTests as withTestsAddon } from '@storybook/addon-jest';
import { ThemeProvider } from 'emotion-theming';
import dhTheme from '@dering-hall/dh-theme';
import testsResults from '../.jest-test-results.json';

const Wrap = ({ index, children }) => (
  <InstantSearch
    appId={process.env.ALGOLIA_ID}
    apiKey={process.env.ALGOLIA_API_KEY}
    indexName={index}
  >
    {children}
  </InstantSearch>
);

Wrap.propTypes = {
  index: PropTypes.string,
  children: PropTypes.node
};

Wrap.defaultProps = {
  index: process.env.ALGOLIA_PRODUCT_INDEX
};

const WrapThemeProvider = ({ children, theme }) => (
  <ThemeProvider theme={theme}>
    <Fragment>
      {children}
    </Fragment>
  </ThemeProvider>
);

WrapThemeProvider.propTypes = {
  theme: PropTypes.object // eslint-disable-line
};

WrapThemeProvider.defaultProps = {
  theme: dhTheme
};

const SearchStateDebug = connectStateResults(({ searchState }) => (
  <div style={{ margin: '1rem 0', fontFamily: 'monospace' }}>
    <pre
      style={{
        background: '#f6f8fa',
        fontSize: '.65rem',
        padding: '.5rem'
      }}
    >
      <strong>searchState</strong> = {JSON.stringify(searchState, null, '\t')}
    </pre>
  </div>
));

const customQueries = {
  small: 'screen and (max-width: 730px)',
  medium: 'screen and (max-width: 970px)',
  large: 'screen and (max-width: 1240px)',
  xlarge: 'screen and (min-width: 1240px)'
};

const MediaQueriesWrapper = ({ children }) => (
  <MediaQueryProvider queries={customQueries}>
    {children}
  </MediaQueryProvider>
);

const OnSmallOnly = withMedia(({ media, children }) => {
  if (!media.small) {
    return null;
  }

  return (
    <Fragment>
      {children}
    </Fragment>
  );
});

const withTests = withTestsAddon({
  results: testsResults
});

export {
  SearchStateDebug,
  MediaQueriesWrapper,
  OnSmallOnly,
  Wrap,
  WrapThemeProvider,
  withTests
};
/* eslint-enable */
