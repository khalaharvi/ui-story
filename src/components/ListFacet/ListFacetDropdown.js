import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _pick from 'lodash.pick';
import { connectHierarchicalMenu } from 'react-instantsearch/connectors';

import { createClassNames } from 'utils';

import PanelCallbackHandler from 'components/Panel/PanelCallbackHandler';
import ListFacetOptions from './ListFacetOptions';
import ListFacetDropdownLink from './ListFacetDropdownLink';

const cx = createClassNames('list-facet');

const itemsPropType = PropTypes.arrayOf(
  PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string,
    count: PropTypes.number.isRequired,
    items: (...args) => itemsPropType(...args)
  })
);

export class ListFacetDropdown extends Component {
  static propTypes = {
    refine: PropTypes.func.isRequired,
    createURL: PropTypes.func.isRequired,
    canRefine: PropTypes.bool.isRequired,
    items: itemsPropType,
    showMore: PropTypes.bool,
    limit: PropTypes.number,
    showMoreLimit: PropTypes.number,
    transformItems: PropTypes.func,
    className: PropTypes.string
  };

  renderItem = (item) => {
    const { createURL, refine } = this.props;

    return (
      <ListFacetDropdownLink
        className={cx('option-link')}
        onClick={() => refine(item.value)}
        href={createURL(item.value)}
      >
        <span className={cx('option-label')}>{item.label}</span>
      </ListFacetDropdownLink>
    );
  };

  render() {
    return (
      <ListFacetOptions
        renderItem={this.renderItem}
        cx={cx}
        {..._pick(this.props, [
          'items',
          'showMore',
          'limit',
          'showMoreLimit',
          'canRefine',
          'className'
        ])}
      />
    );
  }
}

const ListFacetDropdownComponent = props => (
  <PanelCallbackHandler {...props}>
    <ListFacetDropdown {...props} />
  </PanelCallbackHandler>
);

export default connectHierarchicalMenu(ListFacetDropdownComponent);
