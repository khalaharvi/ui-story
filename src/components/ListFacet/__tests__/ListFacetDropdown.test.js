import React from 'react';
import { mountWithTheme } from 'utils/testing';

import { ListFacetDropdown } from '../ListFacetDropdown';

describe('<ListFacetDropdown />', () => {
  it('renders correctly', () => {
    const wrapper = mountWithTheme(
      <ListFacetDropdown
        refine={() => null}
        createURL={() => '#'}
        items={[
          {
            value: 'white',
            count: 10,
            label: 'white',
            items: [
              { value: 'white1', label: 'white1', count: 3 },
              { value: 'white2', label: 'white2', count: 4 }
            ]
          },
          { value: 'black', count: 20, label: 'black' },
          { value: 'blue', count: 30, label: 'blue' }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with className', () => {
    const wrapper = mountWithTheme(
      <ListFacetDropdown
        refine={() => null}
        createURL={() => '#'}
        items={[
          {
            value: 'white',
            count: 10,
            label: 'white',
            items: [
              { value: 'white1', label: 'white1', count: 3 },
              { value: 'white2', label: 'white2', count: 4 }
            ]
          },
          { value: 'black', count: 20, label: 'black' },
          { value: 'blue', count: 30, label: 'blue' }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
        className="custom-list-facet-dropdown"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('refines its value on change', () => {
    const refine = jest.fn();
    const wrapper = mountWithTheme(
      <ListFacetDropdown
        refine={refine}
        createURL={() => '#'}
        items={[
          {
            value: 'white',
            count: 10,
            label: 'white',
            items: [
              { value: 'white1', label: 'white1', count: 3 },
              { value: 'white2', label: 'white2', count: 4 }
            ]
          },
          { value: 'black', count: 20, label: 'black' },
          { value: 'blue', count: 30, label: 'blue' }
        ]}
        canRefine
      />
    );

    const itemParent = wrapper.find('li.dh-list-facet-option--parent');

    expect(itemParent).toHaveLength(1);

    itemParent
      .find('Link')
      .first()
      .simulate('click');
    expect(refine.mock.calls).toHaveLength(1);
    expect(refine.mock.calls[0][0]).toEqual('white');

    wrapper.unmount();
  });

  it('should respect defined limits', () => {
    const wrapper = mountWithTheme(
      <ListFacetDropdown
        refine={() => null}
        createURL={() => '#'}
        items={[
          { value: 'white', count: 10, label: 'white' },
          { value: 'black', count: 20, label: 'black' },
          { value: 'blue', count: 30, label: 'blue' },
          { value: 'green', count: 30, label: 'green' },
          { value: 'cyan', count: 30, label: 'cyan' }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
      />
    );

    const items = wrapper.find('li');

    expect(items).toHaveLength(2);

    wrapper.find('button').simulate('click');

    expect(wrapper.find('li')).toHaveLength(4);

    wrapper.unmount();
  });

  it('should disable show more when no more item to display', () => {
    const wrapper = mountWithTheme(
      <ListFacetDropdown
        refine={() => null}
        createURL={() => '#'}
        items={[
          { value: 'white', count: 10, label: 'white' },
          { value: 'black', count: 20, label: 'black' }
        ]}
        limit={2}
        showMoreLimit={4}
        showMore
        canRefine
      />
    );

    const items = wrapper.find('li');

    expect(items).toHaveLength(2);

    expect(wrapper.find('button[disabled]')).toBeDefined();

    wrapper.unmount();
  });
});
