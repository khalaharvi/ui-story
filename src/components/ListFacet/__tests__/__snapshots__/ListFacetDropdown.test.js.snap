// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListFacetDropdown /> renders correctly 1`] = `
.emotion-30 {
  position: relative;
}

.emotion-16 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.emotion-16 ul {
  padding-left: 15px;
  margin-bottom: 0;
}

.emotion-18 {
  display: block;
  position: relative;
  color: #6d6d6a;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-18:hover {
  color: #242424;
}

.emotion-18 > a {
  position: relative;
  color: #242424;
}

.emotion-18 > a:after {
  content: "";
  display: block;
  position: absolute;
  top: 50%;
  right: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
  pointer-events: none;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 13 8'%3E%3Cpath fill='none' stroke='%236d6d6a' d='M0 1.044L6.5 7 13 1' /%3E%3C/svg%3E");
  background-repeat: no-repeat;
  min-width: 13px;
  min-height: 8px;
}

.emotion-18 > a + ul {
  display: block;
}

.emotion-2 {
  display: block;
  color: #6d6d6a;
  cursor: pointer;
}

.emotion-2 + ul {
  display: none;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  color: #6d6d6a;
  cursor: pointer;
}

.emotion-0 + ul {
  display: none;
}

.emotion-8 {
  display: block;
  position: relative;
  color: #6d6d6a;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-8:hover {
  color: #242424;
}

.emotion-28 {
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  outline: 0;
  background: none;
  cursor: pointer;
  margin: 0;
  color: #cdcbc5;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-28:hover {
  color: #242424;
}

.emotion-28[disabled] {
  display: none;
}

<ListFacetDropdown
  canRefine={true}
  createURL={[Function]}
  items={
    Array [
      Object {
        "count": 10,
        "items": Array [
          Object {
            "count": 3,
            "label": "white1",
            "value": "white1",
          },
          Object {
            "count": 4,
            "label": "white2",
            "value": "white2",
          },
        ],
        "label": "white",
        "value": "white",
      },
      Object {
        "count": 20,
        "label": "black",
        "value": "black",
      },
      Object {
        "count": 30,
        "label": "blue",
        "value": "blue",
      },
    ]
  }
  limit={2}
  refine={[Function]}
  showMore={true}
  showMoreLimit={4}
>
  <WithTheme(ListFacetOptions)
    canRefine={true}
    cx={[Function]}
    items={
      Array [
        Object {
          "count": 10,
          "items": Array [
            Object {
              "count": 3,
              "label": "white1",
              "value": "white1",
            },
            Object {
              "count": 4,
              "label": "white2",
              "value": "white2",
            },
          ],
          "label": "white",
          "value": "white",
        },
        Object {
          "count": 20,
          "label": "black",
          "value": "black",
        },
        Object {
          "count": 30,
          "label": "blue",
          "value": "blue",
        },
      ]
    }
    limit={2}
    renderItem={[Function]}
    showMore={true}
    showMoreLimit={4}
  >
    <ListFacetOptions
      canRefine={true}
      cx={[Function]}
      items={
        Array [
          Object {
            "count": 10,
            "items": Array [
              Object {
                "count": 3,
                "label": "white1",
                "value": "white1",
              },
              Object {
                "count": 4,
                "label": "white2",
                "value": "white2",
              },
            ],
            "label": "white",
            "value": "white",
          },
          Object {
            "count": 20,
            "label": "black",
            "value": "black",
          },
          Object {
            "count": 30,
            "label": "blue",
            "value": "blue",
          },
        ]
      }
      limit={2}
      renderItem={[Function]}
      showMore={true}
      showMoreLimit={4}
    >
      <Styled(div)
        canRefine={true}
        className="dh-list-facet"
      >
        <div
          className="dh-list-facet emotion-30 emotion-31"
        >
          <Styled(ul)
            canRefine={true}
          >
            <ul
              className="emotion-16 emotion-17"
            >
              <Styled(li)
                className="dh-list-facet-option dh-list-facet-option--parent"
                key="white"
                parent={true}
              >
                <li
                  className="dh-list-facet-option dh-list-facet-option--parent emotion-18 emotion-9"
                >
                  <WithTheme(ListFacetDropdownLink)
                    className="dh-list-facet-option-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <ListFacetDropdownLink
                      className="dh-list-facet-option-link"
                      href="#"
                      onClick={[Function]}
                    >
                      <Link
                        className="dh-list-facet-option-link emotion-2"
                        external={false}
                        href="#"
                        noFollow={false}
                        onClick={[Function]}
                      >
                        <Styled(a)
                          className="dh-link dh-list-facet-option-link emotion-2"
                          onClick={[Function]}
                        >
                          <a
                            className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                            onClick={[Function]}
                          >
                            <span
                              className="dh-list-facet-option-label"
                            >
                              white
                            </span>
                          </a>
                        </Styled(a)>
                      </Link>
                    </ListFacetDropdownLink>
                  </WithTheme(ListFacetDropdownLink)>
                  <Styled(ul)>
                    <ul
                      className="emotion-16 emotion-17"
                    >
                      <Styled(li)
                        className="dh-list-facet-option"
                        key="white1"
                        parent={false}
                      >
                        <li
                          className="dh-list-facet-option emotion-8 emotion-9"
                        >
                          <WithTheme(ListFacetDropdownLink)
                            className="dh-list-facet-option-link"
                            href="#"
                            onClick={[Function]}
                          >
                            <ListFacetDropdownLink
                              className="dh-list-facet-option-link"
                              href="#"
                              onClick={[Function]}
                            >
                              <Link
                                className="dh-list-facet-option-link emotion-2"
                                external={false}
                                href="#"
                                noFollow={false}
                                onClick={[Function]}
                              >
                                <Styled(a)
                                  className="dh-link dh-list-facet-option-link emotion-2"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                                    onClick={[Function]}
                                  >
                                    <span
                                      className="dh-list-facet-option-label"
                                    >
                                      white1
                                    </span>
                                  </a>
                                </Styled(a)>
                              </Link>
                            </ListFacetDropdownLink>
                          </WithTheme(ListFacetDropdownLink)>
                        </li>
                      </Styled(li)>
                      <Styled(li)
                        className="dh-list-facet-option"
                        key="white2"
                        parent={false}
                      >
                        <li
                          className="dh-list-facet-option emotion-8 emotion-9"
                        >
                          <WithTheme(ListFacetDropdownLink)
                            className="dh-list-facet-option-link"
                            href="#"
                            onClick={[Function]}
                          >
                            <ListFacetDropdownLink
                              className="dh-list-facet-option-link"
                              href="#"
                              onClick={[Function]}
                            >
                              <Link
                                className="dh-list-facet-option-link emotion-2"
                                external={false}
                                href="#"
                                noFollow={false}
                                onClick={[Function]}
                              >
                                <Styled(a)
                                  className="dh-link dh-list-facet-option-link emotion-2"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                                    onClick={[Function]}
                                  >
                                    <span
                                      className="dh-list-facet-option-label"
                                    >
                                      white2
                                    </span>
                                  </a>
                                </Styled(a)>
                              </Link>
                            </ListFacetDropdownLink>
                          </WithTheme(ListFacetDropdownLink)>
                        </li>
                      </Styled(li)>
                    </ul>
                  </Styled(ul)>
                </li>
              </Styled(li)>
              <Styled(li)
                className="dh-list-facet-option"
                key="black"
                parent={false}
              >
                <li
                  className="dh-list-facet-option emotion-8 emotion-9"
                >
                  <WithTheme(ListFacetDropdownLink)
                    className="dh-list-facet-option-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <ListFacetDropdownLink
                      className="dh-list-facet-option-link"
                      href="#"
                      onClick={[Function]}
                    >
                      <Link
                        className="dh-list-facet-option-link emotion-2"
                        external={false}
                        href="#"
                        noFollow={false}
                        onClick={[Function]}
                      >
                        <Styled(a)
                          className="dh-link dh-list-facet-option-link emotion-2"
                          onClick={[Function]}
                        >
                          <a
                            className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                            onClick={[Function]}
                          >
                            <span
                              className="dh-list-facet-option-label"
                            >
                              black
                            </span>
                          </a>
                        </Styled(a)>
                      </Link>
                    </ListFacetDropdownLink>
                  </WithTheme(ListFacetDropdownLink)>
                </li>
              </Styled(li)>
            </ul>
          </Styled(ul)>
          <Styled(button)
            className="dh-list-facet-option--show-more"
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="dh-list-facet-option--show-more emotion-28 emotion-29"
              disabled={false}
              onClick={[Function]}
            >
              View 
              More
            </button>
          </Styled(button)>
        </div>
      </Styled(div)>
    </ListFacetOptions>
  </WithTheme(ListFacetOptions)>
</ListFacetDropdown>
`;

exports[`<ListFacetDropdown /> renders correctly with className 1`] = `
.emotion-30 {
  position: relative;
}

.emotion-16 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.emotion-16 ul {
  padding-left: 15px;
  margin-bottom: 0;
}

.emotion-18 {
  display: block;
  position: relative;
  color: #6d6d6a;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-18:hover {
  color: #242424;
}

.emotion-18 > a {
  position: relative;
  color: #242424;
}

.emotion-18 > a:after {
  content: "";
  display: block;
  position: absolute;
  top: 50%;
  right: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
  pointer-events: none;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 13 8'%3E%3Cpath fill='none' stroke='%236d6d6a' d='M0 1.044L6.5 7 13 1' /%3E%3C/svg%3E");
  background-repeat: no-repeat;
  min-width: 13px;
  min-height: 8px;
}

.emotion-18 > a + ul {
  display: block;
}

.emotion-2 {
  display: block;
  color: #6d6d6a;
  cursor: pointer;
}

.emotion-2 + ul {
  display: none;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  color: #6d6d6a;
  cursor: pointer;
}

.emotion-0 + ul {
  display: none;
}

.emotion-8 {
  display: block;
  position: relative;
  color: #6d6d6a;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-8:hover {
  color: #242424;
}

.emotion-28 {
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  outline: 0;
  background: none;
  cursor: pointer;
  margin: 0;
  color: #cdcbc5;
  font-family: Monument Grotesk,sans-serif;
  font-size: 19px;
  line-height: 25px;
}

.emotion-28:hover {
  color: #242424;
}

.emotion-28[disabled] {
  display: none;
}

<ListFacetDropdown
  canRefine={true}
  className="custom-list-facet-dropdown"
  createURL={[Function]}
  items={
    Array [
      Object {
        "count": 10,
        "items": Array [
          Object {
            "count": 3,
            "label": "white1",
            "value": "white1",
          },
          Object {
            "count": 4,
            "label": "white2",
            "value": "white2",
          },
        ],
        "label": "white",
        "value": "white",
      },
      Object {
        "count": 20,
        "label": "black",
        "value": "black",
      },
      Object {
        "count": 30,
        "label": "blue",
        "value": "blue",
      },
    ]
  }
  limit={2}
  refine={[Function]}
  showMore={true}
  showMoreLimit={4}
>
  <WithTheme(ListFacetOptions)
    canRefine={true}
    className="custom-list-facet-dropdown"
    cx={[Function]}
    items={
      Array [
        Object {
          "count": 10,
          "items": Array [
            Object {
              "count": 3,
              "label": "white1",
              "value": "white1",
            },
            Object {
              "count": 4,
              "label": "white2",
              "value": "white2",
            },
          ],
          "label": "white",
          "value": "white",
        },
        Object {
          "count": 20,
          "label": "black",
          "value": "black",
        },
        Object {
          "count": 30,
          "label": "blue",
          "value": "blue",
        },
      ]
    }
    limit={2}
    renderItem={[Function]}
    showMore={true}
    showMoreLimit={4}
  >
    <ListFacetOptions
      canRefine={true}
      className="custom-list-facet-dropdown"
      cx={[Function]}
      items={
        Array [
          Object {
            "count": 10,
            "items": Array [
              Object {
                "count": 3,
                "label": "white1",
                "value": "white1",
              },
              Object {
                "count": 4,
                "label": "white2",
                "value": "white2",
              },
            ],
            "label": "white",
            "value": "white",
          },
          Object {
            "count": 20,
            "label": "black",
            "value": "black",
          },
          Object {
            "count": 30,
            "label": "blue",
            "value": "blue",
          },
        ]
      }
      limit={2}
      renderItem={[Function]}
      showMore={true}
      showMoreLimit={4}
    >
      <Styled(div)
        canRefine={true}
        className="dh-list-facet custom-list-facet-dropdown"
      >
        <div
          className="dh-list-facet custom-list-facet-dropdown emotion-30 emotion-31"
        >
          <Styled(ul)
            canRefine={true}
          >
            <ul
              className="emotion-16 emotion-17"
            >
              <Styled(li)
                className="dh-list-facet-option dh-list-facet-option--parent"
                key="white"
                parent={true}
              >
                <li
                  className="dh-list-facet-option dh-list-facet-option--parent emotion-18 emotion-9"
                >
                  <WithTheme(ListFacetDropdownLink)
                    className="dh-list-facet-option-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <ListFacetDropdownLink
                      className="dh-list-facet-option-link"
                      href="#"
                      onClick={[Function]}
                    >
                      <Link
                        className="dh-list-facet-option-link emotion-2"
                        external={false}
                        href="#"
                        noFollow={false}
                        onClick={[Function]}
                      >
                        <Styled(a)
                          className="dh-link dh-list-facet-option-link emotion-2"
                          onClick={[Function]}
                        >
                          <a
                            className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                            onClick={[Function]}
                          >
                            <span
                              className="dh-list-facet-option-label"
                            >
                              white
                            </span>
                          </a>
                        </Styled(a)>
                      </Link>
                    </ListFacetDropdownLink>
                  </WithTheme(ListFacetDropdownLink)>
                  <Styled(ul)>
                    <ul
                      className="emotion-16 emotion-17"
                    >
                      <Styled(li)
                        className="dh-list-facet-option"
                        key="white1"
                        parent={false}
                      >
                        <li
                          className="dh-list-facet-option emotion-8 emotion-9"
                        >
                          <WithTheme(ListFacetDropdownLink)
                            className="dh-list-facet-option-link"
                            href="#"
                            onClick={[Function]}
                          >
                            <ListFacetDropdownLink
                              className="dh-list-facet-option-link"
                              href="#"
                              onClick={[Function]}
                            >
                              <Link
                                className="dh-list-facet-option-link emotion-2"
                                external={false}
                                href="#"
                                noFollow={false}
                                onClick={[Function]}
                              >
                                <Styled(a)
                                  className="dh-link dh-list-facet-option-link emotion-2"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                                    onClick={[Function]}
                                  >
                                    <span
                                      className="dh-list-facet-option-label"
                                    >
                                      white1
                                    </span>
                                  </a>
                                </Styled(a)>
                              </Link>
                            </ListFacetDropdownLink>
                          </WithTheme(ListFacetDropdownLink)>
                        </li>
                      </Styled(li)>
                      <Styled(li)
                        className="dh-list-facet-option"
                        key="white2"
                        parent={false}
                      >
                        <li
                          className="dh-list-facet-option emotion-8 emotion-9"
                        >
                          <WithTheme(ListFacetDropdownLink)
                            className="dh-list-facet-option-link"
                            href="#"
                            onClick={[Function]}
                          >
                            <ListFacetDropdownLink
                              className="dh-list-facet-option-link"
                              href="#"
                              onClick={[Function]}
                            >
                              <Link
                                className="dh-list-facet-option-link emotion-2"
                                external={false}
                                href="#"
                                noFollow={false}
                                onClick={[Function]}
                              >
                                <Styled(a)
                                  className="dh-link dh-list-facet-option-link emotion-2"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                                    onClick={[Function]}
                                  >
                                    <span
                                      className="dh-list-facet-option-label"
                                    >
                                      white2
                                    </span>
                                  </a>
                                </Styled(a)>
                              </Link>
                            </ListFacetDropdownLink>
                          </WithTheme(ListFacetDropdownLink)>
                        </li>
                      </Styled(li)>
                    </ul>
                  </Styled(ul)>
                </li>
              </Styled(li)>
              <Styled(li)
                className="dh-list-facet-option"
                key="black"
                parent={false}
              >
                <li
                  className="dh-list-facet-option emotion-8 emotion-9"
                >
                  <WithTheme(ListFacetDropdownLink)
                    className="dh-list-facet-option-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <ListFacetDropdownLink
                      className="dh-list-facet-option-link"
                      href="#"
                      onClick={[Function]}
                    >
                      <Link
                        className="dh-list-facet-option-link emotion-2"
                        external={false}
                        href="#"
                        noFollow={false}
                        onClick={[Function]}
                      >
                        <Styled(a)
                          className="dh-link dh-list-facet-option-link emotion-2"
                          onClick={[Function]}
                        >
                          <a
                            className="dh-link dh-list-facet-option-link emotion-0 emotion-1"
                            onClick={[Function]}
                          >
                            <span
                              className="dh-list-facet-option-label"
                            >
                              black
                            </span>
                          </a>
                        </Styled(a)>
                      </Link>
                    </ListFacetDropdownLink>
                  </WithTheme(ListFacetDropdownLink)>
                </li>
              </Styled(li)>
            </ul>
          </Styled(ul)>
          <Styled(button)
            className="dh-list-facet-option--show-more"
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="dh-list-facet-option--show-more emotion-28 emotion-29"
              disabled={false}
              onClick={[Function]}
            >
              View 
              More
            </button>
          </Styled(button)>
        </div>
      </Styled(div)>
    </ListFacetOptions>
  </WithTheme(ListFacetOptions)>
</ListFacetDropdown>
`;
